engine/glfw/lib/android/android_window.c

```diff
@@ -438,22 +447,27 @@ void _glfwPlatformPollEvents( void )
 // Called from the looper thread
 void glfwAndroidPollEvents()
 {
-   int ident;
-   int events;
-   struct android_poll_source* source;
-
-   int timeoutMillis = 0;
-   if (_glfwWin.iconified) {
-       timeoutMillis = 1000 * 3;
-   }
-   while ((ident=ALooper_pollAll(timeoutMillis, NULL, &events, (void**)&source)) >= 0)
-   {
-       timeoutMillis = 0;
-       // Process this event.
-       if (source != NULL) {
-           source->process(_glfwWinAndroid.app, source);
-       }
-   }
+    int ident;
+    int events;
+    struct android_poll_source* source;
+
+    int timeoutMillis = 0;
+    if (_glfwWin.iconified) {
+        timeoutMillis = 1000 * 3;
+    }
+    while ((ident=ALooper_pollAll(timeoutMillis, NULL, &events, (void**)&source)) >= 0)
+    {
+        timeoutMillis = 0;
+        // Process this event.
+        if (source != NULL) {
+            source->process(_glfwWinAndroid.app, source);
+        }
+
+        if (_glfwWinAndroid.app->destroyRequested) {
+            androidDestroyWindow();
+            return;
+        }
```

Comment:
this condition (466-469) is a new piece of code, everything else indentations.

Reply:
I'm assuming that it's ok to exit the app right here then. We don't need to wait for threads? E.g. sound.

Reply:
I'm not sure.
As I understand from code 
>_glfwWin.opened = 0;
is a soft exit.

Our loop continue work but see this flag and finish all the threads.

But I'm not sure if it's fine or not. 
In Android examples they not just exit Looper but exit `android_main()` to continue `android_app_entry()` function in `app_glue` and call `android_app_destroy()`

I can do the same, but that means I should change `args.m_Finished = 1;` to exit our loop in `engine_main()`. that means this logic should be in two places:
1. in `glfwAndroidPollEvents` to exit looper
2. in `engine_main -> while (!args.m_Finished)` to change `args.m_Finished = 1;` and exit loop there

Reply:
I added exit from the loop`engine_main()` loop as well. But still not sure if we should force exit app

Reply:
I guess, with the latest update, you _are_ sure?